flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/offset/SeekBinlogByTimestampListener.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.mysql.source.offset;
+
+import com.github.shyiko.mysql.binlog.BinaryLogClient;
+import com.github.shyiko.mysql.binlog.event.Event;
+import com.github.shyiko.mysql.binlog.event.EventHeader;
+import com.github.shyiko.mysql.binlog.event.EventType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** event listener to seek binlog offset by timestamp . */
+public class SeekBinlogByTimestampListener implements BinaryLogClient.EventListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SeekBinlogByTimestampListener.class);
+
+    private final long seekTimestamp;
+    private final BinaryLogClient client;
+    private final BinlogOffset maxBinlogOffset;
+    private BinlogOffset binlogOffset;
+    private BinlogOffset lastBinlogOffset;
+
+    public SeekBinlogByTimestampListener(
+            long seekTimestamp, BinaryLogClient client, BinlogOffset maxBinlogOffset) {
+        this.seekTimestamp = seekTimestamp;
+        this.client = client;
+        this.maxBinlogOffset = maxBinlogOffset;
+    }
+
+    @Override
+    public void onEvent(Event event) {
```

Comment:
IMHO, the logic should be:

1. Judge whether the first event in the current binlog file has timestamp greater than seekTimestamp. If so, skip the binlog file and head on to the previous one. 
    Starting from MySQL 5.0, the first event is always FORMAT_DESC.
2. We have the timestamp of the first binlog event smaller than the one we are seeking. Continue to seek until the very first event that has timestamp greater than or equal to seekTimestamp
3. Return the corresponding binlog filename and position.